lib/types/object/index.js

```diff
@@ -118,16 +119,42 @@ internals.Object = class extends Any {
                 }
             }
 
-            if (target[rename.from] === undefined) {
+
+            const isRegExRename = rename.from instanceof RegExp;
+
+            if (target[rename.from] === undefined && !isRegExRename) {
                 delete target[rename.to];
             }
+            else if (isRegExRename) {
+                const objectKeys = Object.keys(target);
+
+                for (let j = objectKeys.length; j--;) {
+                    if (rename.from.test(objectKeys[j])) {
+                        target[rename.to] = target[objectKeys[j]];
+
+                        // Using regex: Key needs to be normalized to proper case
+
+                        for (let k = this._inner.children.length; k--;) {
```

Comment:
This part is wrong, rename has a`to` argument, use that, plain and simple, don't try to be clever by looking for the matching child. And also make use of the options, some cases are missing here.